d2bs/kolbot/libs/bots/BoBot.js

```diff
@@ -0,0 +1,230 @@
+/**
+*	@filename	BoBot.js
+*	@author		casualBotter
+*	@desc		Meant to run alongside Enchant to Bo players
+*/
+
+function BoBot(){
+	var command, hostile, nick, spot, tick, s, m,
+		startTime = getTickCount(),
+		shitList = [],
+		greet = [];
+
+		this.Bo = function (nick) {
+		if (!Misc.inMyParty(nick)) {
+			say("Accept party invite, noob.");
+
+			return false;
+		}
+
+		var partyUnit,
+			unit = getUnit(0, nick);
+
+		if (getDistance(me, unit) > 10) {
+			say("Get closer.");
+
+			return false;
+		}
+
+		if (!unit) {
+			partyUnit = getParty(nick);
+
+			// wait until party area is readable?
+
+			if ([1,40, 75, 103, 109].indexOf(partyUnit.area) > -1) {
+				say("Cannot Bo in town.");
+				print("cannot bo in town");
+				return false
+			} else{
+				print(partyUnit.area); //for debug
+				Pather.useWaypoint(partyUnit.area);
+				unit = getUnit(0, nick);
+			}
+		}
+
+		if (unit) {
+			do {
+				if (!unit.dead) { // player is alive
+					if (getDistance(me, unit) >= 10) {
+						say("You went too far away.");
+
+						return false;
+					}
+					Precast.doPrecast(true);
+				}
+			} while (unit.getNext());
+		} else {
+			say("Couldn't find you, champ.");
+		}
+
+		unit = getUnit(1);
+
+		if (unit) {
+			do {
+				if (unit.getParent() && unit.getParent().name === nick) { // merc or any other owned unit
+					Precast.doPrecast(true);
+				}
+			} while (unit.getNext());
+		}

```

Comment:
Is this part really needed ? Merc and summons should have bo once you cast it on the player. Imagine if a player has 10 summons the barb will end up casting bo/bc/shout 11 times in a row.

Reply:
Also right i got rid of this.